examples/pytorch/gin/model.py

```diff
@@ -0,0 +1,258 @@
+        self.learn_eps = learn_eps
+        self.eps = nn.Parameter(torch.zeros(self.num_layers-1))
+
+        self.label_idx = label_idx
+        self.label_dim = label_dim
+        self.feature_type = feature_type
+        self.feature_idx = feature_idx
+        self.feature_dim = feature_dim
+        self.hidden_dim = hidden_dim
+        self.output_dim = output_dim
+
+        self.mlps = torch.nn.ModuleList()
+
+        # Batch Norms to be applied on final layer
+        self.batch_norms = torch.nn.ModuleList()
+
+        for layer in range(self.num_layers-1):
+            if layer == 0:
+                self.mlps.append(MLP(num_mlp_layers, input_dim, hidden_dim, hidden_dim))
+            else:
+                self.mlps.append(MLP(num_mlp_layers, hidden_dim, hidden_dim, hidden_dim))
+
+            self.batch_norms.append(nn.BatchNorm1d(hidden_dim))
+
+        # Linear Function: hidden rep -> prediction score
+        self.linears_prediction = torch.nn.ModuleList()
+        for layer in range(num_layers):
+            if layer == 0:
+                self.linears_prediction.append(nn.Linear(input_dim, output_dim))
+            else:
+                self.linears_prediction.append(nn.Linear(hidden_dim, output_dim))
+    
+    def next_layer(self, h, layer, features, nb_counts, num_nodes):
+        pooled_rep = self.mlps[layer](h)
+        h = self.batch_norms[layer](pooled_rep)
+        h = F.relu(h)
+        return h
+
+    def get_score(self, context: dict):
+        num_nodes = len(context['nb_counts'][0])
+        nb_counts = context["nb_counts"].squeeze()
+        features = context["features"].squeeze()
+
+        offset = 0
+        pooled = torch.zeros(num_nodes, self.feature_dim)
+
+        for node_id in range(num_nodes):
+            num_neighbors = nb_counts[node_id].int().item()
+
+            # Aggregate and sum features across all neighbors 
+            neighbor_features = features[offset: offset + num_neighbors]
+            sum_features = neighbor_features.sum(0)
+
+            # Move offset forward to read next set of nb features
+            offset += (num_neighbors + 1)
+
+            # Write to pooled matrix
+            if self.neighbor_pooling_type == "average":
+                sum_features /= num_neighbors
+
+            # Write to pooled matrix
+            pooled[node_id] = sum_features
+
+        # get_logger().info("NB COUNTS: " + str(nb_counts))
+
+        hidden_rep = [pooled]
+        h = pooled
+        
+        for layer in range(self.num_layers - 1):
+            h = self.next_layer(h, layer, features, nb_counts, num_nodes)
+            hidden_rep.append(h)
+
+        score = 0
+        
+        for layer, h in enumerate(hidden_rep):
+            # pooled_h = torch.spmm(nb_features, h)
+            # get_logger().info("Layer " + str(layer) + " | h shape: " + str(pooled_h.shape))
+            score += F.dropout(self.linears_prediction[layer](h), self.final_dropout, training = self.training)
```

Comment:
You might be able to use an Aggregator to handle the pooling here "        ###create sum or average pooling sparse matrix over entire nodes in each graph (num graphs x num nodes)
" 
I'm pretty sure pooling is important for learning